lib/ansible/plugins/inventory/kubevirt.py

```diff
@@ -0,0 +1,263 @@
+        self.inventory.add_group(name)
+        self.inventory.add_group(namespace_group)
+        self.inventory.add_child(name, namespace_group)
+        self.inventory.add_group(namespace_vms_group)
+        self.inventory.add_child(namespace_group, namespace_vms_group)
+        for vm in obj.items:
+            if not (vm.status and vm.status.interfaces):
+                continue
+
+            # Find interface by its name:
+            if interface_name is None:
+                interface = vm.status.interfaces[0]
+            else:
+                interface = next(
+                    (i for i in vm.status.interfaces if i.name == interface_name),
+                    None
+                )
+
+            # If interface is not found or IP address is not reported skip this VM:
+            if interface is None or interface.ipAddress is None:
+                continue
+
+            vm_name = name_format.format(namespace=vm.metadata.namespace, name=vm.metadata.name, uid=vm.metadata.uid)
+            vm_ip = interface.ipAddress
+            vm_annotations = {} if not vm.metadata.annotations else dict(vm.metadata.annotations)
+
+            self.inventory.add_host(vm_name)
+
+            if vm.metadata.labels:
+                # create a group for each label_value
+                for key, value in vm.metadata.labels:
+                    group_name = 'label_{0}_{1}'.format(key, value)
+                    group_name = self._sanitize_group_name(group_name)
+                    self.inventory.add_group(group_name)
+                    self.inventory.add_child(group_name, vm_name)
+                vm_labels = dict(vm.metadata.labels)
+            else:
+                vm_labels = {}
+
+            self.inventory.add_child(namespace_vms_group, vm_name)
+
+            # add hostvars
+            self.inventory.set_variable(vm_name, 'ansible_host', vm_ip)
+            self.inventory.set_variable(vm_name, 'labels', vm_labels)
+            self.inventory.set_variable(vm_name, 'annotations', vm_annotations)
+            self.inventory.set_variable(vm_name, 'object_type', 'vm')
+            self.inventory.set_variable(vm_name, 'resource_version', vm.metadata.resourceVersion)
+            self.inventory.set_variable(vm_name, 'uid', vm.metadata.uid)
+
+            # Add all variables which are listed in 'ansible' annotation:
+            annotations_data = json.loads(vm_annotations.get(annotation_variable, "{}"))
+            for k, v in annotations_data.items():
+                self.inventory.set_variable(vm_name, k, v)
+
+    def verify_file(self, path):
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(('kubevirt.yml', 'kubevirt.yaml')):
+                return True
+        return False
+
+    def _sanitize_group_name(self, name):
+        """ Replace unsupported charactes in group name by underscore character """
+        name = name.replace('/', '_')
```

Comment:
Rather than overwriting the base method from BaseInventoryPlugin, why don't you remove this method? Then your calls to _sanitize_group_name will use a standard regex to better address all possible invalid characters.